omod/src/main/java/org/openmrs/notification/web/controller/AlertFormController.java

```diff
@@ -119,6 +119,8 @@ protected ModelAndView processFormSubmission(HttpServletRequest request, HttpSer
 						if (!userIds.contains(userId)) {
 							recipientsToRemove.add(recipient);
 						}
+						//To remove the users that have already been scheduled
+						userIds.remove(userId);
```

Comment:
@Am-Coder done on this,have you tested this with a running instance of openmrs?

Reply:
Yup, I have verified the working using OpenMRS SDK on both distribution and platform server. Also tested this on OpenMRS-core separately by making use of the generated omod of legacy-ui module.

Reply:
I find your addition confusing because it is under a block for removing all recipients not in the userIds list. Which has nothing to do with clearing the userIds list.

Reply:
**userIds** hold the id's of users that will be called for scheduling as shown [here](https://github.com/openmrs/openmrs-module-legacyui/blob/df54970ee7bde1c3c0913bc600a27398f8aef614/omod/src/main/java/org/openmrs/notification/web/controller/AlertFormController.java#L129) . In [this](https://github.com/openmrs/openmrs-module-legacyui/blob/df54970ee7bde1c3c0913bc600a27398f8aef614/omod/src/main/java/org/openmrs/notification/web/controller/AlertFormController.java#L117) loop where I have added the change, all the recipients of alert are the recipients that have already been scheduled. These id's need to be removed from **userIds** , otherwise you will try to persist the duplicate and hence raise an exception. 

That is why I added the change in the loop.

Reply:
Shouldn't it be in some sort of else clause?

Reply:
Else clause will be more appropriate. Made the changes.

Reply:
@dkayiwa , are some more changes needed for this PR ?